### PR TITLE
Add Smart-ID support

### PR DESCRIPTION
**Pull request by metsma**
SKEID-228

Signed-off-by: Raul Metsma <raul@metsma.ee>

### CODE REVIEW

**Comment by metsma** [comment]
Requires certificate in Central configuration

    -----BEGIN CERTIFICATE-----
    MIIE0zCCA7ugAwIBAgIQbQr/Ky22GFhYWS3oQoJkyTANBgkqhkiG9w0BAQsFADBt
    MQswCQYDVQQGEwJFRTEiMCAGA1UEChMZQVMgU2VydGlmaXRzZWVyaW1pc2tlc2t1
    czEhMB8GA1UECxMYU2VydGlmaXRzZWVyaW1pc3RlZW51c2VkMRcwFQYDVQQDEw5L
    TEFTUzMtU0sgMjAxMDAeFw0xNjEyMjAxMzEwMTlaFw0yMDAxMTkxMzEwMTlaMHQx
    ETAPBgNVBAgMCEhhcmp1bWFhMRAwDgYDVQQHDAdUYWxsaW5uMQswCQYDVQQGEwJF
    RTEiMCAGA1UECgwZQVMgU2VydGlmaXRzZWVyaW1pc2tlc2t1czEcMBoGA1UEAwwT
    cnAtYXBpLnNtYXJ0LWlkLmNvbTCCASIwDQYJKoZIhvcNAQEBBQADggEPADCCAQoC
    ggEBAKgSmOPu6QrndaeJ8dcGs85iqkjTdtd3xiucdxsD/kaqH1X2rD3ZO+5e7jDn
    LQCDUyCC7xRfbtVwZXB4e903IE68Z7Gi9X5oZob7G+4F+gHfrIcCIlFg4/27hBwX
    To1a2Obe0+EKQVD+9Ki/B1L7+ZoN1U1baDKewESWkeqUwH+pLjFmHKnQWhaX2s0T
    F/gUeYlxKV2PYEXZVcnbTFxM8rL6JF6xlTtHMyS69uAoqG1wC9NDBqk0w2vMVadz
    SqBwRUmQB7nCIJEnF4WSxtaG6+hDjT+NHR320vY5ZOrvbiwqBEcEyrv62MzCPmLT
    nyr0IZdXQXrRvlCAkUvEit83KdkCAwEAAaOCAWYwggFiMAkGA1UdEwQCMAAwTwYD
    VR0gBEgwRjAwBgkrBgEEAc4fBwIwIzAhBggrBgEFBQcCARYVaHR0cHM6Ly93d3cu
    c2suZWUvY3BzMAgGBgQAj3oBBzAIBgZngQwBAgIwEwYDVR0lBAwwCgYIKwYBBQUH
    AwEwHwYDVR0jBBgwFoAUXXUUEYz0pY5Cj3uyQESj7tZ6O3IwDgYDVR0PAQH/BAQD
    AgWgMB0GA1UdDgQWBBQVs7nOwNcEPGFWbMs90fu8jXi+3DAeBgNVHREEFzAVghNy
    cC1hcGkuc21hcnQtaWQuY29tMH8GCCsGAQUFBwEBBHMwcTAhBggrBgEFBQcwAYYV
    aHR0cDovL29jc3Auc2suZWUvc3NsMEwGCCsGAQUFBzAChkBodHRwczovL3NrLmVl
    L3VwbG9hZC9maWxlcy9LTEFTUzMtU0tfMjAxMF9FRUNDUkNBX1NIQTM4NC5kZXIu
    Y3J0MA0GCSqGSIb3DQEBCwUAA4IBAQAki4YZ2zwctDre9fZG5OppBXigO6PGn6kk
    Jep4iDY7FOU7ooTB903aydnvkI9fHUWs3fGJLNLDYRIDsHiI7eTnmsB/R8aUFpK3
    +l2YDZ60n5LPLL+uQ3f+wgO+9BagJwGj140EXjR/ac3rZfVb5Pk1RUCv5YzVUcOj
    /WRjCH/HeuJt7lflpgG5Ehlz4iJidFW7bPc5EPRCewGg4+KW3CsHCODZxrzQ1M6B
    Y7XCi78Zggq1XI4qG4xw8zhNNKIqxUB5+tSBvfu1usKzErJ66ZqYcStlIVmU48d1
    aqGJoZ2Litg6bWOO37/0y9fUjRUoY+GriyNWy6GVaxsO5889swUT
    -----END CERTIFICATE-----

File: client/MainWindow.cpp [hunk 1]
```diff
@@ -517,32 +528,51 @@ void MainWindow::buttonClicked( int button )
 		if( !access.validate() )
 			break;
 
-		if( infoSignCard->isChecked() )
+		switch(infoTypeGroup->checkedId())
+		{
+		case 0:
```
Comment (martinpaljak) [inline client/MainWindow.cpp]: What are these magic constants, 0 and 1?

Comment (metsma) [inline client/MainWindow.cpp]: Line 84 85, Token type selection (SmartCard, Mobile-ID, SmartID)

**Comment by martinpaljak** [comment]
Please add significant independent features wrapped with #ifded and relevant config option

**Comment by metsma** [comment]
qdigidoc is feature complete

**Pull request closed without merging**